src/main/java/uk/gov/hmcts/reform/jps/domain/JudicialOfficeHolder.java

```diff
@@ -30,4 +38,35 @@ public class JudicialOfficeHolder {
     @Column(name = "personal_code")
     private String personalCode;
 
+    @ToString.Exclude
+    @LazyCollection(LazyCollectionOption.FALSE)
+    @OneToMany(mappedBy = "judicialOfficeHolder",
+        cascade = {CascadeType.PERSIST, CascadeType.REMOVE})
+    private final List<JohPayroll> johPayrolls = new ArrayList<>();
+
+    public void addJohPayroll(JohPayroll johPayroll) {
+        this.johPayrolls.add(johPayroll);
+        johPayroll.setJudicialOfficeHolder(this);
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+
+        if (!(o instanceof JudicialOfficeHolder that)) {
+            return false;
+        }
+
+        return new EqualsBuilder().append(id, that.id).append(personalCode, that.personalCode).append(
+            johPayrolls.size(),
+            that.johPayrolls.size()
+        ).isEquals();
+    }
+
+    @Override
+    public int hashCode() {
+        return new HashCodeBuilder(17, 37).append(id).append(personalCode).append(johPayrolls).toHashCode();
```

Comment:
This could be replaced by - java.util.Objects.hash()

Reply:
done